src/context.rs

```diff
@@ -183,6 +183,14 @@ impl ToPeer {
     }
 }
 
+impl Drop for ToPeer {
```

Comment:
Why re-arranging this - dtor usually comes last in the traits applied ? Anyway this is producing unnecessary diff